hooks/useEffectDebugger.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * From https://stackoverflow.com/a/59843241/592760
+ */
+import { useRef, useEffect } from 'react';
+
+const usePrevious = (value, initialValue) => {
+  const ref = useRef(initialValue);
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+};
+
+// eslint-disable-next-line import/no-unused-modules
+export const useEffectDebugger = (effectHook, dependencies, dependencyNames: string[] = []): void => {
```

Comment:
useful tool for discovering what was causing all our re-renders. check out the stack overflow link at the top for a great answer.